portfolio/src/main/webapp/tasksApi.js

```diff
@@ -0,0 +1,203 @@
+var GoogleAuth;
+
+// Scope for read access to Tasks API
+const SCOPE_TASKS_READ = 'https://www.googleapis.com/auth/tasks.readonly';
+
+const TASKS_CLIENT_ID =
+    '499747085593-hvi6n4kdrbbfvcuo1c9a9tu9oaf62cr2.apps.googleusercontent.com';
+
+const TASKS_DISCOVERY_DOCS =
+    'https://www.googleapis.com/discovery/v1/apis/tasks/v1/rest';
+
+var TASKS_API_KEY = '';
+
+function fetchApiKey() {
+  fetch('./appConfigServlet').then(response => {
+    TASKS_API_KEY = response.json()['API_KEY'];
+  });
+}
+
+function handleClientLoadTasks() {
+  // Load the API's client and auth2 modules.
+  // Call the initClient function after the modules load.
+  gapi.load('client:auth2', initClientTasks);
+}
+
+function initClientTasks() {
+  fetchApiKey();
+
+  // Initialize the gapi.client object, which app uses to make API requests.
+  // Get API key and client ID from API Console.
+  // 'scope' field specifies space-delimited list of access scopes.
+  gapi.client
+      .init({
+        apiKey: TASKS_API_KEY,
+        clientId: TASKS_CLIENT_ID,
+        discoveryDocs: [TASKS_DISCOVERY_DOCS],
+        scope: SCOPE_TASKS_READ
+      })
+      .then(function() {
+        GoogleAuth = gapi.auth2.getAuthInstance();
+
+        // Listen for sign-in state changes.
+        GoogleAuth.isSignedIn.listen(updateSigninStatus);
+
+        // Handle initial sign-in state. (Determine if user is already signed
+        // in.)
+        updateSigninStatus(GoogleAuth.isSignedIn.get());
+      });
+}
+
+function handleAuthClick() {
+  clearAuthErrorPrompt();
+  if (GoogleAuth.isSignedIn.get()) {
+    // User is authorized and has clicked "Sign out" button.
+    GoogleAuth.signOut();
+  } else {
+    // User is not signed in. Start Google auth flow.
+    GoogleAuth.signIn().catch(error => {
+      handleTaskAuthError(error);
+    });
+  }
+}
+
+function updateSigninStatus(isSignedIn) {
+  if (isSignedIn) {
+    drawImportMenu();
+  } else {
+    clearImportMenu();
+  }
+  updateButtonText(isSignedIn);
+}
+
+/**
+ * Iterate through all the user's tasklists and pass
+ * them to updateTaskList().
+ */
+function importAllTasks() {
+  if (!GoogleAuth.isSignedIn.get()) {
+    return;
+  }
+
+  gapi.client.tasks.tasklists.list({maxResults: 100}).then(function(response) {
+    var taskLists = response.result.items;
+
+    // Check that the variable exists so that no error is thrown.
+    if (taskLists) {
+      taskLists.forEach(tasklist => {
+        importTasklist(tasklist.id);
+      });
+    }
+  });
+}
+
+/** Import a single tasklist identified by its id. */
+function importTasklist(tasklistId) {
+  gapi.client.tasks.tasks
+      .list({tasklist: tasklistId, maxResults: 100, showCompleted: false})
+      .then(function(taskResponse) {
+        var tasks = taskResponse.result.items;
+        if (tasks) {
+          tasks.forEach(task => {
+            const newTask = new Task(task.title, task.notes, '60', '3');
```

Comment:
I think the constructor should take integers for the 3rd and 4th params.

Reply:
Done.